src/overlays/actors/ovl_En_Okarina_Tag/z_en_okarina_tag.c

```diff
@@ -30,14 +29,123 @@ const ActorInit En_Okarina_Tag_InitVars = {
+    f32 xzRange;
+    f32 yRange;
+    s16 yDiff;
+    u16 var_v1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Okarina_Tag/func_8093E68C.s")
+    if (this->unk14C >= 0) {
+        if (this->unk148 == 0) {
+            if (Flags_GetSwitch(play, this->unk14C) != 0) {
+                return;
+            }
+        }
+        if (this->unk148 == 1) {
+            if (Flags_GetSwitch(play, this->unk14C) == 0) {
+                return;
+            }
+        }
+    }
+    var_v1 = this->unk14A;
+    if (var_v1 == 6) {
+        var_v1 = 0xA;
+        if (gSaveContext.save.unk_F65 == 0) {
+            return;
+        }
+    }
+    if (this->unk14A == -1) {
+        var_v1 = 0;
+    }
+    if (func_800B8718(&this->actor, &play->state) != 0) {
+        func_80152434(play, var_v1 + 0x29);
+        this->actionFunc = func_8093E68C;
+    } else {
+        yDiff = ABS_ALT((s16)(this->actor.yawTowardsPlayer - this->actor.world.rot.y));
+        if (yDiff >= 0x4300) {
+            this->unk150 = 0;
+            return;
+        }
+        xzRange = this->unk154;
+        this->unk150 = 1;
+        if (xzRange == 0.0f) {
+            xzRange = 50000.0f;
+        }
+        yRange = this->unk158;
+        if (yRange == 0.0f) {
+            yRange = 50000.0f;
+        }
+        func_800B874C(&this->actor, play, xzRange, yRange);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Okarina_Tag/EnOkarinaTag_Update.s")
+void func_8093E68C(EnOkarinaTag* this, PlayState* play) {
+    if (play->msgCtx.ocarinaMode == 4) {
+        this->actionFunc = func_8093E518;
+    } else {
+        if ((play->msgCtx.ocarinaMode == 3) ||
+            ((this->unk14A == -1) &&
+             ((play->msgCtx.ocarinaMode == 5) || (play->msgCtx.ocarinaMode == 6) || (play->msgCtx.ocarinaMode == 7) ||
+              (play->msgCtx.ocarinaMode == 8) || (play->msgCtx.ocarinaMode == 0xA) || (play->msgCtx.ocarinaMode == 9) ||
+              (play->msgCtx.ocarinaMode == 0xF)))) {
+            if (this->unk14C >= 0) {
+                switch (this->unk148) {
+                    case 0:
+                        Flags_SetSwitch(play, this->unk14C);
+                        break;
+                    case 1:
+                        Flags_UnsetSwitch(play, this->unk14C);
+                        break;
+                    case 2:
+                        if (Flags_GetSwitch(play, this->unk14C) != 0) {
+                            Flags_UnsetSwitch(play, this->unk14C);
+                        } else {
+                            Flags_SetSwitch(play, this->unk14C);
+                        }
+                        break;
+                }
+            }
+            play->msgCtx.ocarinaMode = 4;
+            play_sound(NA_SE_SY_CORRECT_CHIME);
+            this->actionFunc = func_8093E518;
+        }
+    }
+}
+
+void EnOkarinaTag_Update(Actor* thisx, PlayState* play) {
+    EnOkarinaTag* this = THIS;
+    this->actionFunc(this, play);
```

Comment:
```suggestion
    EnOkarinaTag* this = THIS;
    
    this->actionFunc(this, play);
```
This project likes to add one line of whitespace between variable declarations and code statements.